### PR TITLE
dev-python/flask-login: Added ebuild for v0.4.1.

### PR DESCRIPTION
**Pull request by seifertm**
Closes: https://bugs.gentoo.org/611258

### CODE REVIEW

**Comment by sbraz** [comment]
Can you try to add docs and examples? I'll make a PR to address the fact that their PyPI tarballs still don't contains tests and docs despite https://github.com/maxcountryman/flask-uploads/issues/2.

**Comment by sbraz** [comment]
You can probably link to https://github.com/maxcountryman/flask-login/pull/393 above SRC_URI as a reminder when upstream merges it.

**Comment by sbraz** [comment]
Looks like the doc is using intersphinx:
```
WARNING: intersphinx inventory 'https://docs.python.org/3/objects.inv' not fetchable due to <class 'requests.exceptions.ConnectionError'>: ('intersphinx inventory %r not fetchable due to %s: %s', 'https://docs.python.org/3/objects.inv', <class 'requests.exceptions.ConnectionError'>,
```
I've added PyPy3 support to semver in 4ae7bfbc61a3dad05b16885b60e3465fa4bed8e2, I'll merge your PR soon after adding PyPy3 and few small changes.

File: dev-python/flask-login/flask-login-0.4.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+inherit distutils-r1
+
+DESCRIPTION="Login session support for Flask"
+HOMEPAGE="https://pypi.org/project/Flask-Login/"
+# Should be replaced with PyPi URI for next release, if possible
+# See https://github.com/maxcountryman/flask-login/pull/393
+SRC_URI="https://github.com/maxcountryman/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+# pypi tarball is missing tests
+# see https://github.com/maxcountryman/flask-uploads/issues/2
```
Comment (sbraz) [inline dev-python/flask-login/flask-login-0.4.1.ebuild]: This is my fault, I linked to the wrong project at first

File: dev-python/flask-login/flask-login-0.4.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc test"
+
+RDEPEND=">=dev-python/flask-0.10[${PYTHON_USEDEP}]"
+DEPEND="${RDEPEND}
```
Comment (sbraz) [inline dev-python/flask-login/flask-login-0.4.1.ebuild]: RDEPEND is only required at build time when we run tests, it should be moved to `test?`

File: dev-python/flask-login/flask-login-0.4.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	test? (
+		dev-python/nose[${PYTHON_USEDEP}]
+		dev-python/mock[${PYTHON_USEDEP}]
+		dev-python/blinker[${PYTHON_USEDEP}]
+		dev-python/semantic_version[${PYTHON_USEDEP}]
+		$(python_gen_cond_dep 'dev-python/unittest2[${PYTHON_USEDEP}]' 'python2*' pypy)
+	)"
+
+python_compile_all() {
+	use doc && emake -C docs html
```
Comment (sbraz) [inline dev-python/flask-login/flask-login-0.4.1.ebuild]: I tend to set HTML_DOCS here so I can keep the default python_install_all

File: dev-python/flask-login/flask-login-0.4.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	doc? (
+		dev-python/sphinx[${PYTHON_USEDEP}]
+	)
+	test? (
+		dev-python/nose[${PYTHON_USEDEP}]
+		dev-python/mock[${PYTHON_USEDEP}]
+		dev-python/blinker[${PYTHON_USEDEP}]
+		dev-python/semantic_version[${PYTHON_USEDEP}]
+		$(python_gen_cond_dep 'dev-python/unittest2[${PYTHON_USEDEP}]' 'python2*' pypy)
```
Comment (sbraz) [inline dev-python/flask-login/flask-login-0.4.1.ebuild]: This is only for Python 2.6 and older, we don't need it any more:
> unittest2 is a backport of the new features added to the unittest testing framework in Python 2.7 and onwards

**Comment by seifertm** [comment]
Great! Thanks for merging and for the pointers!

**Pull request closed without merging**